### PR TITLE
fix: native-image support feature class

### LINKED ISSUE #685: GraalVM native-image compilation errors
My note about resolving the native-image compilation errors found https://github.com/googleapis/java-compute/pull/682#issuecomment-1106852462.

Note that there's a flaky test `ITPaginationTest ` (https://github.com/googleapis/java-compute/issues/683) where the "Kokoro - Test: Java GraalVM Native Image" fails **before** executing the tests on GraalVM native-image. It outputs "Zone us-central1-a was not found in zones list". If you see this message, it means the invocation hasn't reached native-image.

Pull request for investigation: https://github.com/googleapis/java-compute/pull/682

# NoSuchMethodException: com.google.api.FieldBehavior.valueOf

```
       Suppressed: java.lang.ExceptionInInitializerError
         com.google.cloud.compute.v1.Compute.<clinit>(Compute.java:18809)
         com.google.cloud.compute.v1.Operation.getDescriptor(Operation.java:291)
         com.google.cloud.compute.v1.stub.HttpJsonRegionInstanceGroupManagersStub.<clinit>(HttpJsonRegionInstanceGroupManagersStub.java:84)
         com.google.cloud.compute.v1.RegionInstanceGroupManagersClientTest.startStaticServer(RegionInstanceGroupManagersClientTest.java:58)
         java.lang.reflect.Method.invoke(Method.java:566)
         [...]
       Caused by: java.lang.RuntimeException: Generated message class "com.google.api.FieldBehavior" missing method "valueOf".
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1973)
         com.google.protobuf.GeneratedMessage.access$1100(GeneratedMessage.java:61)
         com.google.protobuf.GeneratedMessage$GeneratedExtension.<init>(GeneratedMessage.java:1789)
         com.google.protobuf.GeneratedMessage.newFileScopedGeneratedExtension(GeneratedMessage.java:1643)
         com.google.api.FieldBehaviorProto.<clinit>(FieldBehaviorProto.java:55)
         [...]
       Caused by: java.lang.NoSuchMethodException: com.google.api.FieldBehavior.valueOf(com.google.protobuf.Descriptors$EnumValueDescriptor)
         java.lang.Class.getMethod(DynamicHub.java:1114)
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1970)
         [...]
```

Protobuf's getMethodOrDie uses reflection:

```
  private static Method getMethodOrDie(
      final Class clazz, final String name, final Class... params) {
    try {
      return clazz.getMethod(name, params);
    } catch (NoSuchMethodException e) {
      throw new RuntimeException(
        "Generated message class \"" + clazz.getName() +
        "\" missing method \"" + name + "\".", e);
    }
  }
```

`com.google.cloud.compute.v1.Compute`'s line 18809 uses `com.google.api.FieldBehaviorProto`.

Screen Shot 2022-04-25 at 10 44 25 AM

`com.google.api.FieldBehaviorProto` calls `GeneratedMessage.newFileScopedGeneratedExtension`.

Screen Shot 2022-04-25 at 10 46 16 AM
 
Note that `<clinit>` for `com.google.cloud.compute.v1.Compute` means it's static field initialization for the class, not a constructor. This explains other errors for "NoClassDefFoundError: Could not initialize class com.google.cloud.compute.v1.Compute".

# NoSuchMethodException: com.google.cloud.OperationResponseMapping.valueOf

The same error as previous:

```
       Caused by: java.lang.RuntimeException: Generated message class "com.google.cloud.OperationResponseMapping" missing method "valueOf".
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1973)
         com.google.protobuf.GeneratedMessage.access$1100(GeneratedMessage.java:61)
         com.google.protobuf.GeneratedMessage$GeneratedExtension.<init>(GeneratedMessage.java:1789)
         com.google.protobuf.GeneratedMessage.newFileScopedGeneratedExtension(GeneratedMessage.java:1643)
         com.google.cloud.ExtendedOperationsProto.<clinit>(ExtendedOperationsProto.java:52)
         [...]
       Caused by: java.lang.NoSuchMethodException: com.google.cloud.OperationResponseMapping.valueOf(com.google.protobuf.Descriptors$EnumValueDescriptor)
         java.lang.Class.getMethod(DynamicHub.java:1114)
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1970)
         [...]
```

# NoSuchMethodException: com.google.protobuf.DescriptorProtos$FieldOptions.getCtype

```
       Suppressed: java.lang.ExceptionInInitializerError
         com.google.cloud.compute.v1.Operation.getDescriptor(Operation.java:291)
         com.google.cloud.compute.v1.stub.HttpJsonNodeGroupsStub.<clinit>(HttpJsonNodeGroupsStub.java:78)
         com.google.cloud.compute.v1.NodeGroupsClientTest.startStaticServer(NodeGroupsClientTest.java:59)
         java.lang.reflect.Method.invoke(Method.java:566)
         org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
         [...]
       Caused by: java.lang.RuntimeException: Generated message class "com.google.protobuf.DescriptorProtos$FieldOptions" missing method "getCtype".
         com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1847)
         com.google.protobuf.GeneratedMessageV3.access$1000(GeneratedMessageV3.java:79)
         com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$SingularFieldAccessor$ReflectionInvoker.<init>(GeneratedMessageV3.java:2176)
         com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$SingularFieldAccessor.<init>(GeneratedMessageV3.java:2248)
         com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$SingularEnumFieldAccessor.<init>(GeneratedMessageV3.java:2743)
         [...]
       Caused by: java.lang.NoSuchMethodException: com.google.protobuf.DescriptorProtos$FieldOptions.getCtype()
         java.lang.Class.getMethod(DynamicHub.java:1114)
         com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1844)
         [...]
```

# Generated message class "com.google.protobuf.DescriptorProtos$FieldOptions$Builder" missing method "getCtype"

```
Failures (121):
  JUnit Vintage:BackendServicesClientTest
    ClassSource [className = 'com.google.cloud.compute.v1.BackendServicesClientTest', filePosition = null]
    => org.opentest4j.MultipleFailuresError: Multiple Failures (2 failures)
	java.lang.ExceptionInInitializerError: <no message>
	java.lang.NullPointerException: <no message>
       org.junit.vintage.engine.execution.TestRun.getStoredResultOrSuccessful(TestRun.java:196)
       org.junit.vintage.engine.execution.RunListenerAdapter.fireExecutionFinished(RunListenerAdapter.java:226)
       org.junit.vintage.engine.execution.RunListenerAdapter.testRunFinished(RunListenerAdapter.java:93)
       org.junit.runner.notification.SynchronizedRunListener.testRunFinished(SynchronizedRunListener.java:42)
       org.junit.runner.notification.RunNotifier$2.notifyListener(RunNotifier.java:103)
       [...]
       Suppressed: java.lang.ExceptionInInitializerError
         com.google.cloud.compute.v1.Operation.getDescriptor(Operation.java:291)
         com.google.cloud.compute.v1.stub.HttpJsonBackendServicesStub.<clinit>(HttpJsonBackendServicesStub.java:74)
         com.google.cloud.compute.v1.BackendServicesClientTest.startStaticServer(BackendServicesClientTest.java:59)
         java.lang.reflect.Method.invoke(Method.java:566)
         org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
         [...]
       Caused by: java.lang.RuntimeException: Generated message class "com.google.protobuf.DescriptorProtos$FieldOptions$Builder" missing method "getCtype".
         com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1847)
         com.google.protobuf.GeneratedMessageV3.access$1000(GeneratedMessageV3.java:79)
         com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$SingularFieldAccessor$ReflectionInvoker.<init>(GeneratedMessageV3.java:2177)
         com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$SingularFieldAccessor.<init>(GeneratedMessageV3.java:2248)
         com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$SingularEnumFieldAccessor.<init>(GeneratedMessageV3.java:2743)
         [...]
       Caused by: java.lang.NoSuchMethodException: com.google.protobuf.DescriptorProtos$FieldOptions$Builder.getCtype()
         java.lang.Class.getMethod(DynamicHub.java:1114)
         com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1844)
         [...]
```

# Bad Request: The resource ... is not ready

This is not a GraalVM native-image failure.

```
Failures (1):
  JUnit Vintage:ITSmokeInstancesTest:testCapitalLetterField
    MethodSource [className = 'com.google.cloud.compute.v1.integration.ITSmokeInstancesTest', methodName = 'testCapitalLetterField', methodParameterTypes = '']
    => java.util.concurrent.ExecutionException: com.google.api.gax.rpc.InvalidArgumentException: Bad Request
       com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:588)
       com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:567)
       com.google.common.util.concurrent.FluentFuture$TrustedFuture.get(FluentFuture.java:91)
       com.google.common.util.concurrent.ForwardingFuture.get(ForwardingFuture.java:66)
       com.google.api.gax.longrunning.OperationFutureImpl.get(OperationFutureImpl.java:125)
       [...]
     Caused by: com.google.api.gax.rpc.InvalidArgumentException: Bad Request
       com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:52)
       com.google.api.gax.httpjson.HttpJsonApiExceptionFactory.createApiException(HttpJsonApiExceptionFactory.java:76)
       com.google.api.gax.httpjson.HttpJsonApiExceptionFactory.create(HttpJsonApiExceptionFactory.java:54)
       com.google.api.gax.httpjson.HttpJsonExceptionCallable$ExceptionTransformingFuture.onFailure(HttpJsonExceptionCallable.java:97)
       com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:67)
       [...]
     Caused by: com.google.api.client.http.HttpResponseException: 400 Bad Request
DELETE [https://compute.googleapis.com:443/compute/v1/projects/gcloud-devel/global/firewalls/gapic-fw-rule9cff16a2](https://compute.googleapis.com/compute/v1/projects/gcloud-devel/global/firewalls/gapic-fw-rule9cff16a2)
{
  "error": {
    "code": 400,
    "message": "The resource 'projects/gcloud-devel/global/firewalls/gapic-fw-rule9cff16a2' is not ready",
    "errors": [
      {
        "message": "The resource 'projects/gcloud-devel/global/firewalls/gapic-fw-rule9cff16a2' is not ready",
        "domain": "global",
        "reason": "resourceNotReady"
      }
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #685 

I didn't see the ending of the error message approach, where I update the list and wait 30 min to get error and repeat. It already reached ~500 classes (https://github.com/googleapis/java-compute/commit/0425a99c87614bbcef37aac0a64ecff6bd87f28e). So I try registering all classes in the proto directory for reflection:

### CODE REVIEW

**Comment by suztomo** [comment]
```
java.lang.AssertionError: Zone us-central1-a was not found in zones list.
	at com.google.cloud.compute.v1.integration.ITPaginationTest.testPaginationIterating(ITPaginationTest.java:122)
```

**Comment by suztomo** [comment]
https://source.cloud.google.com/results/invocations/8c0474b8-e199-4faf-b2a4-683ef232f65f/targets/cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-compute%2Fpresubmit%2Fgraalvm-native/log this seems true positive:

```
Failures (121):
  JUnit Vintage:RegionInstanceGroupManagersClientTest
    ClassSource [className = 'com.google.cloud.compute.v1.RegionInstanceGroupManagersClientTest', filePosition = null]
    => org.opentest4j.MultipleFailuresError: Multiple Failures (2 failures)
	java.lang.ExceptionInInitializerError: <no message>
	java.lang.NullPointerException: <no message>
       org.junit.vintage.engine.execution.TestRun.getStoredResultOrSuccessful(TestRun.java:196)
       org.junit.vintage.engine.execution.RunListenerAdapter.fireExecutionFinished(RunListenerAdapter.java:226)
       org.junit.vintage.engine.execution.RunListenerAdapter.testRunFinished(RunListenerAdapter.java:93)
       org.junit.runner.notification.SynchronizedRunListener.testRunFinished(SynchronizedRunListener.java:42)
       org.junit.runner.notification.RunNotifier$2.notifyListener(RunNotifier.java:103)
       [...]
       Suppressed: java.lang.ExceptionInInitializerError
         com.google.cloud.compute.v1.Compute.<clinit>(Compute.java:18809)
         com.google.cloud.compute.v1.Operation.getDescriptor(Operation.java:291)
         com.google.cloud.compute.v1.stub.HttpJsonRegionInstanceGroupManagersStub.<clinit>(HttpJsonRegionInstanceGroupManagersStub.java:84)
         com.google.cloud.compute.v1.RegionInstanceGroupManagersClientTest.startStaticServer(RegionInstanceGroupManagersClientTest.java:58)
         java.lang.reflect.Method.invoke(Method.java:566)
         [...]
       Caused by: java.lang.RuntimeException: Generated message class "com.google.api.FieldBehavior" missing method "valueOf".
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1973)
         com.google.protobuf.GeneratedMessage.access$1100(GeneratedMessage.java:61)
         com.google.protobuf.GeneratedMessage$GeneratedExtension.<init>(GeneratedMessage.java:1789)
         com.google.protobuf.GeneratedMessage.newFileScopedGeneratedExtension(GeneratedMessage.java:1643)
         com.google.api.FieldBehaviorProto.<clinit>(FieldBehaviorProto.java:55)
         [...]
       Caused by: java.lang.NoSuchMethodException: com.google.api.FieldBehavior.valueOf(com.google.protobuf.Descriptors$EnumValueDescriptor)
         java.lang.Class.getMethod(DynamicHub.java:1114)
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1970)
         [...]
       Suppressed: java.lang.NullPointerException
         com.google.cloud.compute.v1.RegionInstanceGroupManagersClientTest.stopServer(RegionInstanceGroupManagersClientTest.java:73)
         java.lang.reflect.Method.invoke(Method.java:566)
         org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
         org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
         org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
         [...]
```

```
       Suppressed: java.lang.NoClassDefFoundError: Could not initialize class com.google.cloud.compute.v1.Compute
         com.google.cloud.compute.v1.Operation.getDescriptor(Operation.java:291)
         com.google.cloud.compute.v1.stub.HttpJsonRoutersStub.<clinit>(HttpJsonRoutersStub.java:75)
         com.google.cloud.compute.v1.RoutersClientTest.startStaticServer(RoutersClientTest.java:59)
         java.lang.reflect.Method.invoke(Method.java:566)
         org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
         [...]
       Suppressed: java.lang.NullPointerException
         com.google.cloud.compute.v1.RoutersClientTest.stopServer(RoutersClientTest.java:73)
         java.lang.reflect.Method.invoke(Method.java:566)
         org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
         org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
         org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
         [...]
```

Retrying.

**Comment by suztomo** [comment]
"Kokoro - Test: Java GraalVM Native Image" succeeded.

File: google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java [hunk 1]
```diff
@@ -0,0 +1,3657 @@
...
+        "com.google.cloud.compute.v1.TestIamPermissionsInstanceTemplateRequest");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.TestIamPermissionsInstanceTemplateRequest$Builder");
+    computeClassesBuilder.add("com.google.cloud.compute.v1.TestIamPermissionsLicenseCodeRequest");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.TestIamPermissionsLicenseCodeRequest$Builder");
+    computeClassesBuilder.add("com.google.cloud.compute.v1.TestIamPermissionsLicenseRequest");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.TestIamPermissionsLicenseRequest$Builder");
+    computeC
```
Comment (mpeddada1) [inline google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java:3627]: Out of curiosity, what happens when we remove these lines?

Comment (suztomo) [reply google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java:3627]: You get the errors such as:

```
       Suppressed: java.lang.ExceptionInInitializerError
         com.google.cloud.compute.v1.Compute.<clinit>(Compute.java:18809)
         com.google.cloud.compute.v1.Operation.getDescriptor(Operation.java:291)
         com.google.cloud.compute.v1.stub.HttpJsonRegionInstanceGroupManagersStub.<clinit>(HttpJsonRegionInstanceGroupManagersStub.java:84)
         com.google.cloud.compute.v1.RegionInstanceGroupManagersClientTest.startStaticServer(RegionInstanceGroupManagersClientTest.java:58)
         java.lang.reflect.Method.invoke(Method.java:566)
         [...]
       Caused by: java.lang.RuntimeException: Generated message class "com.google.api.FieldBehavior" missing method "valueOf".
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1973)
         com.google.protobuf.GeneratedMessage.access$1100(GeneratedMessage.java:61)
         com.google.protobuf.GeneratedMessage$GeneratedExtension.<init>(GeneratedMessage.java:1789)
         com.google.protobuf.GeneratedMessage.newFileScopedGeneratedExtension(GeneratedMessage.java:1643)
         com.google.api.FieldBehaviorProto.<clinit>(FieldBehaviorProto.java:55)
         [...]
       Caused by: java.lang.NoSuchMethodException: com.google.api.FieldBehavior.valueOf(com.google.protobuf.Descriptors$EnumValueDescriptor)
         java.lang.Class.getMethod(DynamicHub.java:1114)
         com.google.protobuf.GeneratedMessage.getMethodOrDie(GeneratedMessage.java:1970)
         [...]
```

https://github.com/googleapis/java-compute/issues/685

File: google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java [hunk 2]
```diff
@@ -0,0 +1,3657 @@
...
+    // https://github.com/googleapis/java-compute/issues/685
+    ImmutableList.Builder<String> computeClassesBuilder = ImmutableList.builder();
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesInstanceGroupManagerRequest");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesInstanceGroupManagerRequest$Builder");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesRegionInstanceGroupManagerRequest");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesRegionInstanceGroupManagerRequest$Builder");
```
Comment (mpeddada1) [inline google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java:56]: Could you point me to the tests that fail if we don't include these `com.google.cloud.compute.v1` classes? I've experienced situations where I had to add a ton a classes for reflection to make the test work but it was just one class that was needed in the end, so I'm secretly hoping that we'll run into same scenario here.

Comment (suztomo) [reply google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java:56]: https://source.cloud.google.com/results/invocations/81073aee-a5fa-4581-af46-462972101862/targets/cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-compute%2Fpresubmit%2Fgraalvm-native/log

Here is an example:

```
  JUnit Vintage:NetworkFirewallPoliciesClientTest:getTest
    MethodSource [className = 'com.google.cloud.compute.v1.NetworkFirewallPoliciesClientTest', methodName = 'getTest', methodParameterTypes = '']
    => com.google.api.gax.rpc.UnknownException: java.lang.RuntimeException: Generated message class "com.google.cloud.compute.v1.FirewallPolicy" missing method "getAssociationsList".
       com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:119)
       com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:41)
       com.google.api.gax.httpjson.HttpJsonApiExceptionFactory.create(HttpJsonApiExceptionFactory.java:67)
       com.google.api.gax.httpjson.HttpJsonExceptionCallable$ExceptionTransformingFuture.onFailure(HttpJsonExceptionCallable.java:97)
       com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:67)
       [...]
       Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
         com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
         com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
         com.google.cloud.compute.v1.NetworkFirewallPoliciesClient.get(NetworkFirewallPoliciesClient.java:687)
         com.google.cloud.compute.v1.NetworkFirewallPoliciesClient.get(NetworkFirewallPoliciesClient.java:662)
         com.google.cloud.compute.v1.NetworkFirewallPoliciesClientTest.getTest(NetworkFirewallPoliciesClientTest.java:388)
         [...]
     Caused by: java.lang.RuntimeException: Generated message class "com.google.cloud.compute.v1.FirewallPolicy" missing method "getAssociationsList".
       com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1847)
       com.google.protobuf.GeneratedMessageV3.access$1000(GeneratedMessageV3.java:79)
       com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$RepeatedFieldAccessor$ReflectionInvoker.<init>(GeneratedMessageV3.java:2409)
       com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$RepeatedFieldAccessor.<init>(GeneratedMessageV3.java:2479)
       com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$RepeatedMessageFieldAccessor.<init>(GeneratedMessageV3.java:3004)
       [...]
     Caused by: java.lang.NoSuchMethodException: com.google.cloud.compute.v1.FirewallPolicy.getAssociationsList()
       java.lang.Class.getMethod(DynamicHub.java:1114)
       com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1844)
       [...]
```

File: google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (suztomo) [reply google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java:3624]: With that approach, we need to exclude the inner classes. It's an additional step in listing classes. I think it's easier to maintain the class list if we just register all classes regardless of nested or not.

What do you think?

**Pull request closed without merging**